### ISSUE TITLE
[pickers] Integral support for for reduced motion

### ISSUE THREAD

**Issue by thany** [mui/mui-x#9740]
### Duplicates

- [X] I have searched the existing issues

### Latest version

- [X] I have tested the latest version

### Steps to reproduce 🕹

Link to live example: https://mui.com/x/react-date-pickers/getting-started/

Steps:
1. Make sure reduced motion is enabled on your OS or browser.
2. Open the date picker and move the month around

### Current behavior 😯

I see animation, and become disoriented. People with more severe motion sensitivity may experience headaches and/or nausia.

### Expected behavior 🤔

No animation when `prefers-reduced-motion` is set.

### Context 🔦

I think this is basic accessibility, and our project needs to meet certain accessibility standards.

We could workaround it, but I feel if a library provides animations pretty much hardcoded, it's on them to deal with disabling them when the browser/OS requests to do so.

### Your environment 🌎

Not applicable

**Comment by LukasTy** [comment]
Thank you for opening this issue! 🙏 
The mentioned shortcoming is real and should be addressed. 👌

**Comment by oliviertassinari** [comment]
Linked in https://github.com/mui/material-ui/issues/16367.

**Comment by oliviertassinari** [comment]
I'm not sure this is solved, two example of animation that aren't "reduced":

https://github.com/mui/mui-x/assets/3165635/6d4405f4-c703-4d3b-b2ce-f1ea40817460

https://deploy-preview-9823--material-ui-x.netlify.app/x/react-date-pickers/#overview

**Comment by LukasTy** [comment]
Thank you for spotting the Digital Clock animation, which was not accounted for. 💯 
In regards to the opening animation—that's an interesting point to also look into. 💡

**Comment by thany** [comment]
Interesting that macOS seemingly keeps smooth scrolling enabled even though reduced motion is enabled... Is that what you meant while showing the time picker, @oliviertassinari? Because that's an animation controlled by the OS, not by MUI. Unless there's something else I didn't notice.

**Comment by LukasTy** [comment]
> Because that's an animation controlled by the OS, not by MUI. Unless there's something else I didn't notice.

I'm pretty sure that it is a browser-level animation and a feature that browsers implement on their own.
Firefox does disable the `smooth-scrolling` when `reduce motion` is toggled on MacOS, but Chrome and Safari don't. We'll wrap the `smooth` scrolling rule in a media query on our end to avoid this inconsistency. 👌
